### DISCUSSION TITLE
Github Pages files give 404 >50% of the time

### DISCUSSION THREAD

**Discussion by rongjie-alter** [community/community#23639] [Other Feature Feedback, Questions, & Ideas]
<p>I have a Github Pages project site at <a href="https://github.com/liveahero-wiki/cdn/tree/gh-pages" rel="noopener nofollow ugc">https://github.com/liveahero-wiki/cdn/tree/gh-pages</a> .</p>
<p>However, the files (especially the JSON files) give 404 errors most of the time. Yes, you hear me right, most of the time, but sometime it does return the file content.</p>
<p>PNG files seem to have higher probability of returning successfully: like <code>https://liveahero-wiki.github.io/cdn/Texture2D/fg_rakta_h01_Atlas0.png</code></p>
<p>JSON files fail most of the time, like <code>https://liveahero-wiki.github.io/cdn/MonoBehaviour/fg_huckle_s01.json</code> and <code>https://liveahero-wiki.github.io/cdn/Monobehaviour/fg_mokdai_h01_skin1.json</code></p>
<p>It was deployed more than 10 hours ago, I don’t know how much longer should I wait if Github wants to call it a caching issue.</p>

**Comment by otter-computer** [comment]
<p>Hi <a class="mention" href="/rongjie-alter">@rongjie-alter</a>, welcome to the GitHub Support Community! I’m not seeing any issues with your Pages site on our side—everything seems to be normal so your pages site should be available.</p>
<p>It could be something to do with your local network, or the network route used to reach the Pages servers, that are causing the issue. Would you be able to give some more specific information on how exactly this is happening, on what devices, on what networks, and when?</p>
<p><strong>An important note:</strong> I notice the name of this repository is <code>cdn</code> and it appears to be filled with image assets and configuration files. GitHub Pages is intended to be a free web host for static sites for personal and organization projects only. It is against the GitHub Pages terms of service to use Pages as a CDN. If you need a service to host assets for an external project then I’d recommend moving your project to a dedicated CDN or static file hosting service such as Azure Storage or AWS S3. If you are using Pages as a CDN for an external project your repository may be disabled among other things.</p>

**Reply by GrantGryczan** [reply]
I found this post while looking for information on whether using Pages as a static file host for external sites is allowed. If that's against the terms, could you point me to where the terms say that? I'm unable to find any mention of this.

Similarly, I'm also wondering if the same rule applies to raw.githubusercontent.com, as I'm unable to find anything in the terms prohibiting that too.

**Comment by rongjie-alter** [comment, marked as answer]
<p>The issue was sorted out the other day, sorry for not updating it here. Basically I used “Monobehaviour” instead of “MonoBehaviour” in the url, that was why it wasn’t working.</p>
<p>The project name <code>/cdn</code> may be misleading, they are static files that the org project site (<a href="https://github.com/liveahero-wiki/liveahero-wiki.github.io" rel="noopener nofollow ugc">https://github.com/liveahero-wiki/liveahero-wiki.github.io</a>) will need.</p>
<p>The reason why we decouple it into a separate project is because putting them in org project make the Jekyll building slower (since our Github actions step will need to pull more files (directly commit into org project or as submodule)) and org project site (which is a game wiki) updates in a much faster pace (a few times a day) than these static files (about one or twice a month).</p>
<p>This does not count as violating Github’s ToS right?</p>

**Comment by otter-computer** [comment]
<p>Ah, good to hear you’ve figured it out already!</p>
<p>And thanks for the clarification! Yes, using it as a source for another site hosted on GitHub Pages should be fine. It’s only really an issue if you’re using it for an <strong>externally-hosted</strong> project like an app, browser extension, etc. :sweat_smile:</p>
